### ISSUE TITLE
Hooks are counted as tests

### ISSUE THREAD

**Issue by yovasx2** [avajs/ava#2142]
[<img alt="Issuehunt badges" src="https://img.shields.io/badge/IssueHunt-%2450%20Funded-%2300A156.svg" />](https://issuehunt.io/r/avajs/ava/issues/2142)

When a hook fails it increases the count of tests:
```
not ok 23 - api › users › createUser › [C806006, monitoring, validation] Verify a user is created
not ok 25 - api › users › createUser › Delete test user for [C806006, monitoring, validation] Verify a user is created
```
**Delete test user** is a hook and it counted as a test, the current number of tests is 24, not 25

this is happening for verbose and tap formats

---

<details>
<summary>
<b>IssueHunt Summary</b>
</summary>

### Backers (Total: $50.00)

- [<img src='https://avatars3.githubusercontent.com/u/44827199?v=4' alt='issuehunt' width=24 height=24> issuehunt](https://issuehunt.io/u/issuehunt) ($50.00)
### Submitted pull Requests
- [#2178 do not report failing hooks as test](https://issuehunt.io/r/avajs/ava/pull/2178)
---

### Tips

- Checkout the [Issuehunt explorer](https://issuehunt.io/r/avajs/ava/) to discover more funded issues.
- Need some help from other developers? [Add your repositories](https://issuehunt.io/r/new) on IssueHunt to raise funds.
---
IssueHunt has been backed by the following sponsors. [Become a sponsor](https://issuehunt.io/membership/members)
</details>

**Comment by novemberborn** [comment]
Nice find, we should fix that.

**Comment by mikaoelitiana** [comment]
I am interested to work on this issue

**Comment by mikaoelitiana** [comment]
@yovasx2 can you please provide a reproduction test with failing hook?

**Comment by mikaoelitiana** [comment]
@yovasx2 @novemberborn what would be the expected behaviour? Should the line be removed at all or should we display `not ok 24 - api › users › createUser › Delete test user for [C806006, monitoring, validation] Verify a user is created` ?

**Comment by novemberborn** [comment]
> what would be the expected behaviour? Should the line be removed at all or should we display `not ok 24 - api › users › createUser › Delete test user for [C806006, monitoring, validation] Verify a user is created` ?

I don't really understand TAP… @yovasx2?

**Comment by yovasx2** [comment]
Each test case is in a line starting with `ok` or `not ok`, but hooks, in this case, `beforeEach` is displayed as a test since the line is starting with `not ok`, so you have 2 tests in your example but TAP is showing 3 (not ok), please check this for more detail: https://testanything.org/

**Comment by yovasx2** [comment]
> @yovasx2 @novemberborn what would be the expected behaviour? Should the line be removed at all or should we display `not ok 24 - api › users › createUser › Delete test user for [C806006, monitoring, validation] Verify a user is created` ?

I think it should be output as a comment, not a failing test

**Comment by mikaoelitiana** [comment]
@yovasx2 unfortunately I couldn't find a direct reference to the test in the `hook-failed` event so it's quite difficult to put the hooks comment under each test. 
What I achieved is something like the following but I am not sure it's valid to TAP specification: 

```
TAP version 13
# foo
ok 1 - foo
# bar
ok 2 - bar
---
my after hook for foo failed.
...
---
my after hook for bar failed.
...

1..2
# tests 2
# pass 2
# fail 0
```

**Comment by yovasx2** [comment]
That could lead to some problems since the `extra`s are mixed

**Comment by Rokt33r** [comment]
@sindresorhus It seems to be solved by #2193. Could you confirm to reward in here? https://issuehunt.io/r/avajs/ava/issues/2142
